### PR TITLE
4816 ui improvement

### PR DESCRIPTION
**Pull request by saulhidalgoaular**
Supports #4816

### CODE REVIEW

**Comment by IanMayo** [comment]
@saulhidalgoaular - we've still got a build dependency problem - see Travis.

**Comment by saulhidalgoaular** [comment]
> @saulhidalgoaular - we've still got a build dependency problem - see Travis.

oh, I see, Sorry.

btw, I am hopefully finishing the Search Option Today.

**Comment by IanMayo** [comment]
> > @saulhidalgoaular - we've still got a build dependency problem - see Travis.
> 
> oh, I see, Sorry.

My fix worked :-)

**Comment by helenayele** [comment]
Icons are added on the tree and 'Connection Test' button is now on proper location:
[Image from Gyazo](https://gyazo.com/84a5f466e60eaea627890831838475c8)

**Comment by helenayele** [comment]
Is there other method to close the import button other than selecting close (X) button. It would be nice if there is close button near Import button. Or can we have other 'Import&Close' button. This will import and close the dialog

File: org.mwc.debrief.pepys/src/main/java/org/mwc/debrief/pepys/model/ModelConfiguration.java [hunk 1]
```diff
@@ -92,7 +100,7 @@ public void apply() throws NoSuchMethodException, SecurityException, Instantiati
 	public void doImport() {
 		if (_bridge == null) {
 			doImportProcessMockup(treeModel);
-		}else {
+		} else {
 			doImport(treeModel, new InternTreeItemFiltering() {
```
Comment (IanMayo) [inline org.mwc.debrief.pepys/src/main/java/org/mwc/debrief/pepys/model/ModelConfiguration.java:104]: @saulhidalgoaular  - I could do with a comment explaining what's happening with these two `doImport()` statements. I guess it's to ensure `States` are loaded before `Contacts` - but we should describe that, please.

**Comment by helenayele** [comment]
Full debrief goes to not responding state after importing states/comments.
[Image from Gyazo](https://gyazo.com/4278ed6f23ef2474629f73ccc3b088a8)

**Comment by IanMayo** [comment]
@helenayele - could you make a recording that shows which data was imported?  Maybe the problem is sensitive to which data is being imported.

**Comment by helenayele** [comment]
Still Debrief goes to not responding state when I import Narratives
Steps:
1. Select database import dialog
2. keep the default values for time period and Area parameters
3. Select Contacts and Comments data types
4. Select Apply and Import and Close the dialog.
5. See that Full debrief goes to not-responding state.
https://screencast-o-matic.com/watch/cYfn2L4Civ

**Comment by IanMayo** [comment]
Helen - I see the Time Controller period selector has lots and lots of lines.

So, this may be related to the bug that Ayesha has been working on.  

If you just import a single file, does it work ok?

**Comment by helenayele** [comment]
I tried by selecting the first state selected. But the reponding state of full debrief still has issues. it is on and off. It goes to not responding state for a while then comes back and then turns to not responding state. See this video:
https://screencast-o-matic.com/watch/cYfno94CKa

**Comment by IanMayo** [comment]
aah, @helenayele - that's because there is already a track loaded -  so the Time Controller still has to cover a very long period of time.

If you load a Pepys track into an empty plot - hopefully it will work fine.

**Comment by helenayele** [comment]
Okay, When I import all the available states /comments , on empty plot , full debrief goes to not responding state. And it takes much of the resources (system memory) :
[Image from Gyazo](https://gyazo.com/9bb1ad8085acd701613e259d20addfcf)

**Comment by helenayele** [comment]
The time controller doesn't show the time on the slider for this Track:subject:
[Image from Gyazo](https://gyazo.com/48141683cb0394aca442a4a30f2e74a4)

Otherwise for a single or two tracks , debrief doesn't have performance issue and plays well:
[Image from Gyazo](https://gyazo.com/994e542722929cbcf67ce48433763b9c)

**Comment by IanMayo** [comment]
@helenayele - let's see if those issues go away once we've merged @ayeshare 's PR.

**Comment by saulhidalgoaular** [comment]
> 
> 
> I tried by selecting the first state selected. But the reponding state of full debrief still has issues. it is on and off. It goes to not responding state for a while then comes back and then turns to not responding state. See this video:
> https://screencast-o-matic.com/watch/cYfno94CKa

It looks to me like it is the Time Controller, because Items were imported correctly and it started to be unresponsive after opening the Time Controller was rendered.

**Comment by IanMayo** [comment]
@helenayele - you should move to this PR, please - since it will _hopefully_ fix these issues too.

**Comment by IanMayo** [comment]
Ok, @helenayele - this branch includes the Time Controller fixes.  Hopefully your performance troubles have gone away.

No rush, but I would appreciate getting it done tomorrow morning, please.

**Comment by helenayele** [comment]
Sorry for the late reply but the performance issue is just go away. thanks

**Pull request merged**